proposals/exception-handling/Exceptions.md

```diff
@@ -455,27 +448,51 @@ access to the data fields of a `Exception` if a matching tag is given. This last
 check ensures that without access to a WebAssembly module's exported exception
 tag, the associated data fields cannot be read.
 
+The `Exception` constructor can take an optional `ExceptionOptions` argument,
+which can optionally contain `traceStack` entry. When `traceStack` is `true`,
+web VMs can attach a stack trace string to `Exception.stack` field, as in
+JavaScript's `Error` class. While `Exception` is not a subclass of JavaScript's
+`Error` and it can be used to represent normal control flow constructs,
+`traceStack` field can be set when we use it to represent errors. The format of
+stack trace strings conform to the [WebAssembly stack trace
+conventions](https://webassembly.github.io/spec/web-api/index.html#conventions).
+When `ExceptionOption` is not provided or it does not contain `traceStack`
+entry, `traceStack` is considered `false` by default.
+
+To preserve stack trace info when crossing the JS to Wasm boundary, `Exception`
+can internally contain an optional `externref` value containing a stack trace
+string, which is propagated when caught by `catch` and rethrown by `rethrow`.
```

Comment:
Hm, is this normative text or is it just an implementation note? IIUC, "internally" means that the reference is not observable in Wasm. Does it even matter then whether it is an externref?

Reply:
I suspect it is less error-prone to acknowledge the extra data that needs to be preserved in the core spec, rather than monkey-patching it from the outside. I'm not sure if that means we need to spell out that it's an externref, though.

Reply:
@Ms2ger, I'm not sure how we could do that without actually devolving into spec'ing how JS stack traces work – presumably, the trace also needs to be extended in certain situations. Giving the handwavy-ness of stack traces in JS itself, we can only get that wrong. I'd rather not go there.

Reply:
@rossberg Do you not prefer taking the optional stack trace on/off option in `WebAssembly.Exception` constructor, or you are OK with that part but doesn't want to say anything about how it is achieved and we delegate that entirely to toolchains?

Reply:
@aheejin, the latter. Or at least I would keep the "how" slightly more abstract.
```suggestion
To preserve stack trace info when crossing the JS to Wasm boundary, exceptions
can internally contain a stack trace, which is propagated when caught by `catch` and rethrown by `rethrow`.
```

Reply:
Done in #203.